pandapower/build_branch.py

```diff
@@ -444,10 +452,14 @@ def _calc_branch_values_from_trafo_df(net, ppc, trafo_df=None, sequence=1, updat
 def _calc_r_x_y_from_dataframe(net, trafo_df, vn_trafo_lv, vn_lv, ppc, sequence=1, update_vk_values=True):
     mode = net["_options"]["mode"]
     trafo_model = net["_options"]["trafo_model"]
-
-    r, x = _calc_r_x_from_dataframe(mode, trafo_df, vn_lv, vn_trafo_lv, net.sn_mva,
-                                    sequence=sequence, characteristic=net.get("characteristic"),
-                                    update_vk_values=update_vk_values)
+    if 'tap_dependency_table' in trafo_df:
+        r, x = _calc_r_x_from_dataframe(mode, trafo_df, vn_lv, vn_trafo_lv, net.sn_mva,
```

Comment:
Please check: For me this seems fishy . depending on the existance of net in the function, you calculate tap dependency table? Is one of these possible: pass the table instead of net to be more explicit about the intention, or use a different function name for this.

Reply:
@Fabsit not sure what you mean here. @hilbrich's approach here was to ensure backward compatibility. So if tap_dependency_table is included in the trafo parameters, then it means we are in new version, otherwise use the old logic. Does this make sense?